### PR TITLE
feat: CI to test aztec circuits with current commit of bberg

### PR DESCRIPTION
**Pull request by dbanks12**
# Blockers

* [x] Blocked by https://github.com/AztecProtocol/barretenberg/pull/368
  * [x] Once this ^ is merged, merge/rebase master into this branch
* [x] Blocked by https://github.com/AztecProtocol/aztec3-packages/pull/432
  * [x] Once this is merged, change `cpp/.aztec-packages-commit` to master or a specific commit on master

# Description

* Integration tests of Aztec circuits to ensure that bberg PRs dont break circuits.
* `aztec3-packages` commit to run against can be configured in `cpp/.aztec-packages-commit`

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [x] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [x] There are no circuit changes, OR specifications in `/markdown/specs` have been updated.
- [x] There are no circuit changes, OR a cryptographer has been assigned for review.
- [x] I've updated any terraform that needs updating (e.g. environment variables) for deployment.
- [x] The branch has been rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewer's next convenience.
- [x] New functions, classes, etc. have been documented according to the doxygen comment format. Classes and structs must have `@brief` describing the intended functionality.
- [x] If existing code has been modified, such documentation has been added or updated.

### CODE REVIEW

**Comment by dbanks12** [comment]
Work is done here, just waiting for blocker PR to be merged

File: cpp/scripts/run_aztec_circuits_tests [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  cd /usr/src/; \
+  git clone https://github.com/AztecProtocol/aztec3-packages.git; \
+  cd /usr/src/aztec3-packages/circuits/cpp; \
+  git checkout $AZTEC_COMMIT; \
+  rm -rf /usr/src/aztec3-packages/circuits/cpp/barretenberg;
+  mv /usr/src/barretenberg .; \
+  cmake --preset $PRESET $CONFIGURE_OPTS; \
+  cmake --build --preset $PRESET; \
+  cd /usr/src/aztec3-packages/circuits/cpp/barretenberg/cpp/srs_db; \
+  ./download_ignition.sh $NUM_TRANSCRIPTS; \
```
Comment (ludamad) [inline cpp/scripts/run_aztec_circuits_tests:49]: I think this is differently from how we get the transcript in aztec packages? but I guess we cant use that here

**Review by ludamad — APPROVED** [review]
LGTM

**Review by codygunton — APPROVED** [review]
It's so beautiful

File: cpp/scripts/run_aztec_circuits_tests (hunk 1 above)
Comment (dbanks12) [reply cpp/scripts/run_aztec_circuits_tests:49]: Yep. And I spent a full day trying to reuse docker images from aztec-packages but gave up for now.

**Review by Maddiaa0 — APPROVED** [review]

**Pull request merged**